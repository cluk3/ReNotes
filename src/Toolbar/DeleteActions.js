export const SET_ITEM_TO_DELETE = "SET_ITEM_TO_DELETE";

export function setItemToDelete(entity, id) {
  return {
    type: SET_ITEM_TO_DELETE,
    payload: {
      entity,
      id
    }
  };
}

export function deleteReducer(state = {}, { type, payload = {} }) {
  switch (type) {
    case SET_ITEM_TO_DELETE:
      const { entity, id } = payload;
      return {
        entity,
        id
      };
    default:
      return state;
  }
}
