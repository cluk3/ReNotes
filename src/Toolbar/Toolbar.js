import React, { PureComponent } from 'react';
import styled from 'styled-components';

const ToolbarDiv = styled.div`
  height: 36px;
  background-color: #dcdcdc;
  color: white;
  border-bottom: #a9a9a9 solid 1px;
  display: flex;
  justify-content: flex-start;
  align-items: center;
`;

class Toolbar extends PureComponent {
  render() {
    return <ToolbarDiv>{this.props.children}</ToolbarDiv>;
  }
}

export default Toolbar;
