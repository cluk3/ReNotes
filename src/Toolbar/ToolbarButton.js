import React, { PureComponent } from "react";
import styled from "styled-components";

const Button = styled.button`
  &:active {
    background-color: #dcdcdc;
    outline: none;
  }
  &:hover > svg {
    fill: #dcdcdc;
  }
  margin-right: 0.5em;
  border-radius: 0.5em;
  padding: 0.2em 1.2em;
  min-height: 2.4em;
  outline: none;
`;

export default class ToolbarButton extends PureComponent {
  render() {
    return <Button onClick={this.props.onClick}>{this.props.children}</Button>;
  }
}
