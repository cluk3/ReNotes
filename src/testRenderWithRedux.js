import React from 'react';
import { createStore } from 'redux';
import { Provider, connect } from 'react-redux';
import { render, fireEvent, cleanup } from '@testing-library/react';
import { initialState as notesInitialState } from './Notes/modules/notes';
import { initialState as foldersInitialState } from './Folders/modules/folders';
import { initialState as focusedElementInitialState } from './containers/Focusable/stateManager';

const reducer = state => state;
const defaultInitialState = {
  folders: foldersInitialState,
  notes: notesInitialState,
  focusedElement: focusedElementInitialState
};

export function renderWithRedux(
  ui,
  {
    initialState = defaultInitialState,
    store = createStore(reducer, initialState)
  } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store
  };
}
