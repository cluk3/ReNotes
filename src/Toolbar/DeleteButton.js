import React, { PureComponent } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { deleteFolder } from '../Folders/modules/folders';
import { deleteNoteAndElectNewActive } from '../Notes/modules/notes';
import { ENTITIES } from 'constants.js';
import deleteIcon from '../assets/notes-delete-icon.png';
import ToolbarButton from './ToolbarButton';

const confirmFolderDeleteMessage =
  'Deleting the folder will delete also all the note into it, are you sure?';

class DeleteButton extends PureComponent {
  deleteSelectedItem() {
    const {
      folders,
      notes,
      focusedElement,
      deleteFolder,
      deleteNoteAndElectNewActive
    } = this.props;

    if (focusedElement.elementType === ENTITIES.FOLDERS) {
      const isFolderEmpty =
        folders.byId[folders.activeFolder].notes.length === 0;
      const confirmDelete = !isFolderEmpty
        ? window.confirm(confirmFolderDeleteMessage)
        : true;

      confirmDelete && deleteFolder(folders);
    } else {
      deleteNoteAndElectNewActive(notes.activeNote, folders.activeFolder);
    }
  }
  render() {
    return (
      <ToolbarButton onClick={() => this.deleteSelectedItem()}>
        <img src={deleteIcon} height="18px" alt="delete" />
      </ToolbarButton>
    );
  }
}

function mapStateToProps({ folders, notes, focusedElement }) {
  return {
    folders,
    notes,
    focusedElement
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      deleteFolder,
      deleteNoteAndElectNewActive
    },
    dispatch
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(DeleteButton);
