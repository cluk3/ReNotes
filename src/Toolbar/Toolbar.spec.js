import React from 'react';
import { renderWithRedux } from '../testRenderWithRedux'
import Toolbar from './Toolbar';
import 'jest-styled-components';

test('it works', () => {
    const { container } = renderWithRedux(<Toolbar />);
    expect(container.firstChild).toMatchSnapshot()
})
