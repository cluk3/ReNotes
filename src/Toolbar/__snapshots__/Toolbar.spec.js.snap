// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  height: 36px;
  background-color: #dcdcdc;
  color: white;
  border-bottom: #a9a9a9 solid 1px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
/>
`;
